WalletWasabi/WabiSabi/Models/DecompositionAlgs/Decomposition.cs

```diff
@@ -0,0 +1,36 @@
+using NBitcoin;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class Decomposition : IEnumerable<Money>
+	{
+		public Decomposition(IEnumerable<Money> amounts)
+		{
+			Amounts = amounts.OrderBy(c => c).ToList();
+		}
+
+		private List<Money> Amounts { get; }
+
+		public void AddSorted(Money coin)
+		{
+			int index = Amounts.BinarySearch(coin);
+			if (index < 0)
+			{
+				Amounts.Insert(~index, coin);
+			}
```

Comment:
We could consider using a `SortedList` instead of keeping it sorted manually. If we do that then there is no reason for having this class.

Reply:
You mean `SortedList<Money,Money>`? We are not using SortedList anywhere in the project. I am not sure how it is working, but List with BinarySearch suits our needs.

Reply:
Yes, we have never need to keep a list sorted I think. Anyway, with sorted list or without sorted list this class does nothing valuable and could be absorbed by the one that uses it.

Reply:
We have this https://github.com/molnard/WalletWasabi/blob/fdfd926a4f85922796cef9a807d50d67874bb3f8/WalletWasabi/Extensions/ListExtensions.cs#L56

But I cannot use it because of immutables.

Reply:
Done.